tests/proxy.spec.ts

```diff
@@ -75,6 +76,64 @@ it('should work with IP:PORT notion', async ({browserType, browserOptions, serve
   await browser.close();
 });
 
+it('should not work with localhost by default on Chromium', async ({ browserType, browserOptions }) => {
+  let proxiedRequest = false;
+  const server = net.createServer(socket => {
+    socket.on('data', () => {
+      proxiedRequest = true;
+      socket.end();
+    });
+  });
+  await new Promise(f => server.listen(0, f));
+
+  const opts: LaunchOptions = {
+    ...browserOptions,
+    proxy: { server: `http://127.0.0.1:${(server.address() as any).port}` },
+  };
+
+  const browser = await browserType.launch(opts);
+
+  const page = await browser.newPage();
+
+  await page.goto('http://localhost:9999/request').catch(() => {});
+  await browser.close();
+  server.close();
+  if (browserType.name() === 'chromium')
+    expect(proxiedRequest).toBeFalsy();
+  else
+    expect(proxiedRequest).toBeTruthy();
+});
+
+it('should work with localhost', async ({ browserType, browserOptions }) => {
```

Comment:
not sure if a custom test for that is needed, since we are already testing this behaviour in the `portForwardingServer.spec.ts`

Reply:
I left this test in place as it actually had to get a bit more complicated because of platform quirks. The tests assert what the new documentation claims to be the state of the world.